pydantic_ai_slim/pydantic_ai/_parts_manager.py

```diff
@@ -152,6 +170,127 @@ def handle_text_delta(
             self._parts[part_index] = part_delta.apply(existing_text_part)
             return PartDeltaEvent(index=part_index, delta=part_delta)
 
+    def _handle_text_delta_with_thinking_tags(
+        self,
+        *,
+        vendor_part_id: VendorId | None,
+        content: str,
+        id: str | None = None,
+        thinking_tags: tuple[str, str],
+        ignore_leading_whitespace: bool = False,
+    ) -> ModelResponseStreamEvent | None:
+        """Handle text delta with thinking tag detection and buffering for split tags."""
+        start_tag, end_tag = thinking_tags
+
+        # Combine any buffered content with the new content
+        buffered = self._tag_buffer.get(vendor_part_id, '') if vendor_part_id is not None else ''
+        combined_content = buffered + content
+
+        # Check if we're currently building a thinking part
+        part_index = self._vendor_id_to_part_index.get(vendor_part_id) if vendor_part_id is not None else None
+        in_thinking_mode = part_index is not None and isinstance(self._parts[part_index], ThinkingPart)
+
+        if in_thinking_mode:
+            # Look for the end tag
+            if end_tag in combined_content:
+                # Found complete end tag
+                before_end, after_end = combined_content.split(end_tag, 1)
+
+                # Add any content before the end tag to the thinking part
+                last_event = None
+                if before_end:
+                    last_event = self.handle_thinking_delta(vendor_part_id=vendor_part_id, content=before_end)
+
+                # Close the thinking part
+                self._vendor_id_to_part_index.pop(vendor_part_id)
+                self._tag_buffer.pop(vendor_part_id, None)
+
+                # Process any remaining content after the end tag
+                if after_end:
```

Comment:
As @phemmer said on the other issue, we need to support `foo<think>bar</think>baz` and return 3 events